cli/format.go

```diff
@@ -372,29 +341,46 @@ func applyFormatters(ctx context.Context) func() error {
 
 		// iterate the files channel, checking if any pipeline wants it, and attempting to apply if so.
 		for file := range filesCh {
-			var matches []string
 
-			for key, pipeline := range pipelines {
-				if !pipeline.Wants(file) {
-					continue
+			var sequence []*format.Formatter
+
+			for _, formatter := range formatters {
+				if formatter.Wants(file) {
+					sequence = append(sequence, formatter)
 				}
-				matches = append(matches, key)
-				tryApply(key, file)
 			}
-			switch len(matches) {
-			case 0:
+
+			// sort by priority in ascending order
+			slices.SortFunc(sequence, func(a, b *format.Formatter) int {
+				priorityA := a.Priority()
+				priorityB := b.Priority()
+
+				result := priorityA - priorityB
+				if result == 0 {
+					// formatters with the same priority are sorted lexicographically to ensure a deterministic outcome
+					result = cmp.Compare(a.Name(), b.Name())
+				}
+				return result
+			})
```

Comment:
This concern belong to the tryApply function, or a NewTask function since it's related to the stability of the key